client_listener/src/listener_process.rs

```diff
@@ -106,6 +103,16 @@ impl ListenerProcess {
             }
 
             select! {
+                event = self.event_out_buffer_rx.recv() =>
+                {
+                    let event_sender = Arc::clone(&self.event_sender);
+                    let shutdown_flag = Arc::clone(&self.shutdown_flag);
+                    if let Err(e) = event_sender.send(&event.unwrap()).await {
```

Comment:
Sending here will still block the main task if send() blocks, which brings back the deadlock issue that caused me to spawn sends in the first place.

If we're going to use an unbounded channel to make sure that this task never blocks, then there needs to be a separate task that reads it and forwards to the socket; if we're doing the socket writes in this task then we can do them directly and don't need the extra channel.